app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -76,4 +83,100 @@ class PreferenceController @Inject()(
+             |
+             |   Given I am a customer who has successfully enrolled in ITSA
+             |   When my itsaId is already linked to a different entityId in entity resolver
+             |   Then my itsaId will not be added (i.e linked to) any entityId
+             |   And an error will be generated
+           */
+          }
+        } else {
+          // entity.itsa.nonDefined, which means no link has been created yet
+          if (!authToken_saUtr.isDefined) {
+            /*
+             | Case 1.4 - SAUTR does not exist in the  Auth token
+             |
+             |   Given I am a customer who has successfully enrolled in ITSA
+             |   When SAUTR does not exist in the Auth token
+             |   Then my itsaId will be added (i.e linked to) the entityId
+             */
+            entityResolverConnector
+              .update(resolvedEntity.copy(itsa = Some(passedBack_itsaId)))
+              .map { _ =>
+                Ok("itsaId successfully linked to entityId")
+              }
+          } else {
+            // authToken_saUtr.isDefined
+            if (resolvedEntity.saUtr.isDefined && (resolvedEntity.saUtr.get == authToken_saUtr.get)) {
+              /*
+               | Case 1.1 - SAUTR in Auth token is the same as SAUTR in entity resolver
+               |
+               |   Given I am a customer who has successfully enrolled in ITSA
+               |   When SAUTR in Auth token is the same as SAUTR in entity resolver
+               |   Then my itsaId will be added (i.e linked to) the entityId
+               */
+              entityResolverConnector
+                .update(resolvedEntity.copy(itsa = Some(passedBack_itsaId)))
+                .map { _ =>
+                  Ok("itsaId successfully linked to entityId")
+                }
+            } else {
+              /*
+               | Case 1.2 - SAUTR in Auth token is different from  SAUTR in entity resolver
+               |
+               |   Given I am a customer who has successfully enrolled in ITSA
+               |   When SAUTR in Auth token is different from  SAUTR in entity resolver
+               |   Then my itsaId will not be added (i.e linked to) any entityId
+               |   And an error will be generated
+               */
+              Future.successful(Unauthorized("SAUTR in Auth token is different from SAUTR in entity resolver"))
+
+              // TODO Isn't the following acceptance criteria already included in Case 1.2 ???
+              // Case 1.3 - SAUTR in Auth token is linked to a different entityId  in entity resolver
+              //
+              //   Given I am a customer who has successfully enrolled in ITSA
+              //   When SAUTR in Auth token is linked to a different entityId  in entity resolver
+              //   Then my itsaId will not be added (i.e linked to) any entityId
+              //   And an error will be generated
+            }
+          }
+        }
+      }
```

Comment:
I would suggest using pattern matching and getting rid of the comments, the code should be self explanitory :)

Reply:
Yeah maybe you can just reference the JIRA ticket (or some confluence doc if there is any) in the comment only...

Reply:
Ok, fine.
I will be removing the most of those comments